opm/simulators/flow/partitionCells.cpp

```diff
@@ -40,68 +48,325 @@
 #include <numeric>
 #include <stdexcept>
 #include <string>
+#include <tuple>
 #include <type_traits>
+#include <unordered_map>
+#include <utility>
+#include <vector>
+
+// ===========================================================================
+// Zoltan-based partitioning if availble in this build configuration.
+// ---------------------------------------------------------------------------
+
+#if HAVE_MPI && HAVE_ZOLTAN
 
 namespace {
 
+/// Partition rank's interior cells into non-overlapping domains using the
+/// Zoltan graph partitioning software package.
+class ZoltanPartitioner
+{
+public:
+    /// Constructor.
+    ///
+    /// \tparam GridView DUNE grid view type
+    ///
+    /// \param[in] grid_view Current rank's reachable cells.
+    ///
+    /// \param[in] local_to_global Callback for mapping rank's local cells
+    ///   to globally unique cell IDs.  May for instance return the
+    ///   Cartesian cell ID of each local cell.
+    template <class GridView>
+    explicit ZoltanPartitioner(const GridView&                                          grid_view,
+                               const Opm::ParallelNLDDPartitioningZoltan::GlobalCellID& local_to_global);
+
+    /// Form internal connectivity graph between rank's reachable, interior
+    /// cells.
+    ///
+    /// \tparam GridView DUNE grid view type
+    ///
+    /// \tparam Element Grid view's representation of a cell.
+    ///
+    /// \param[in] grid_view Current rank's reachable cells.
+    ///
+    /// \param[in] wells Collection of wells to which this rank's partition
+    ///   vector must be adapted.  Wells must not intersect multiple local
+    ///   domains/blocks.
+    ///
+    /// \param[in] zoltan_ctrl Control paramerters for on-rank subdomain
+    ///   partitioning.
+    template <class GridView, class Element>
+    void buildLocalGraph(const GridView&                                grid_view,
+                         const std::vector<Opm::Well>&                  wells,
+                         const Opm::ZoltanPartitioningControl<Element>& zoltan_ctrl);
+
+    /// Partition rank's interior cells into non-overlapping domains using
+    /// the Zoltan graph partitioning software pacakge.
```

Comment:
package

Reply:
Thanks for noticing. I've pushed an update to fix that.